### ISSUE TITLE
Generation issue

### ISSUE THREAD

**Issue by fmirault** [MakeAWishFoundation/SwiftyMocky#182]
Hi ! 

I just installed SwiftyMocky in my project (`pod 'SwiftyMocky', '~> 3.2.0-pre'`) I have only one protocol to be mocked for now.
When I run the command line to generate mocks, I have the following error : 

`error: Error Domain=NSCocoaErrorDomain Code=4 "“SwiftTemplate” couldn’t be moved to “Mock.swifttemplate” because either the former doesn’t exist, or the folder containing the latter doesn’t exist." UserInfo={NSSourceFilePathErrorKey=/var/folders/dd/d2prjvc972q3m6gmsqq2w5740000gn/T/SwiftTemplate/0.16.0/.build/debug/SwiftTemplate, NSUserStringVariant=(
    Move
), NSDestinationFilePath=/Users/admin/Library/Caches/Sourcery/Mock.swifttemplate/PBsnqAU8tnnHn%2BxIuc%2FbnhFyXeg6gUoROx%2FnRD5IGG8%3D, NSFilePath=/var/folders/dd/d2prjvc972q3m6gmsqq2w5740000gn/T/SwiftTemplate/0.16.0/.build/debug/SwiftTemplate, NSUnderlyingError=0x7fbde513fc00 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}`

I don't really understand the error, because the destination folder for the generated file exists, do you have any highlight about it ?

**Comment by amichnia** [comment]
@fmirault  Hi, one thing, Could you try:
1. Remove `~/Library/Caches/Sourcery`
2. Run Sourcery it with --disableCache flag?

If that does not help, pleas provide:
- the command to start generation
- version of Sourcery (if you use get_sourcery.sh script)
- version of Xcode / Swift / CommandLine tools

**Comment by torstenlehmann** [comment]
Have the same Issue like described.

I did 1. and when I do 2. I get

```
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'launch path not accessible'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff426cfcf9 __exceptionPreprocess + 256
	1   libobjc.A.dylib                     0x00007fff6d263a17 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff426cfb2b +[NSException raise:format:] + 201
	3   Foundation                          0x00007fff448b599f -[NSConcreteTask launchWithDictionary:error:] + 1291
	4   sourcery                            0x0000000101568313 $SSo6NSTaskC13SourcerySwiftE10runCommand33_97A733547C26B01DA3FEDC2EDE49AED9LL4path9arguments11environment20currentDirectoryPathAC13ProcessResultAELLVSS_SaySSGSDyS2SG0R3Kit0R0VSgtKFZTf4xnnnd_n + 1923
	5   sourcery                            0x000000010156310b $S13SourcerySwift0B8TemplateC6renderySSypKF + 907
	6   sourcery                            0x000000010149d4b2 $S13SourcerySwift0B8TemplateC0A9Framework0C00A0AdEP6renderySS0A7Runtime0C7ContextCKFTW + 66
	7   sourcery                            0x00000001014bd3c9 $S17SourceryFramework9GeneratorO8generate_8template9argumentsSS0A7Runtime5TypesC_AA8Template_pSDySSSo8NSObjectCGtKFZ + 425
	8   sourcery                            0x00000001014a281a $S8SourceryAAC8generate33_DB5C4364BFA94633EE953DD412A08C30LL_16forParsingResult10outputPathSS0A9Framework8Template_p_0A7Runtime5TypesC5types_SaySS4file_SDySSSo8_NSRangeVG6rangesSDyS2SG12indentationstG12inlineRangest0N3Kit0N0VtKF + 826
	9   sourcery                            0x00000001014b2cf2 $S8SourceryAAC8generate33_DB5C4364BFA94633EE953DD412A08C30LL6source13templatePaths6output13parsingResultyAA6SourceO_AA0L0VAA6OutputV0A7Runtime5TypesC5types_SaySS4file_SDySSSo8_NSRangeVG6rangesSDyS2SG12indentationstG12inlineRangesttKFy0A9Framework8Template_pKXEfU_Tf4nnxn_n + 98
	10  sourcery                            0x00000001014b32d1 $S8SourceryAAC8generate33_DB5C4364BFA94633EE953DD412A08C30LL6source13templatePaths6output13parsingResultyAA6SourceO_AA0L0VAA6OutputV0A7Runtime5TypesC5types_SaySS4file_SDySSSo8_NSRangeVG6rangesSDyS2SG12indentationstG12inlineRangesttKFTf4dxnnnn_n + 881
	11  sourcery                            0x00000001014b46c6 $S8SourceryAAC12processFiles_14usingTemplates6output10forceParseSayAA13FolderWatcherO5LocalCGSgAA6SourceO_AA5PathsVAA6OutputVSaySSGtKF0A7Runtime5TypesC5types_SaySS4file_SDySSSo8_NSRangeVG6rangesSDyS2SG12indentationstG12inlineRangestANKcfU1_Tf4nnnxn_n + 3542
	12  sourcery                            0x00000001014b5663 $S8SourceryAAC12processFiles_14usingTemplates6output10forceParseSayAA13FolderWatcherO5LocalCGSgAA6SourceO_AA5PathsVAA6OutputVSaySSGtKFTf4nxnnn_n + 1843
	13  sourcery                            0x000000010147c048 $S8Sourcery6runCLIyyFySb_S4bSay7PathKit0D0VGA3f2ESaySSGAgEtcfU_ + 2232
	14  sourcery                            0x000000010147c680 $SS5bSay7PathKit0A0VGA3d2CSaySSGAeCs5Error_pIegyyyyyggggnnggnzo_S5bA4d2c2eCsAF_pIegnnnnnnnnnnnnnnzo_TR032$S8Sourcery6runCLIyyFySb_S4bSay7A27Kit0D0VGA3f2ESaySSGAgEtcfU_Tf3nnnnnnnnnnnnnnpf_n + 80
	15  sourcery                            0x00000001013e7346 $S9Commander7commandyAA11CommandType_px_q_q0_q1_q2_q3_q4_q5_q6_q7_q8_q9_q10_q11_y05ValueD0Qz_ADQy_ADQy0_ADQy1_ADQy2_ADQy3_ADQy4_ADQy5_ADQy6_ADQy7_ADQy8_ADQy9_ADQy10_ADQy11_tKctAA18ArgumentDescriptorRzAaSR_AaSR0_AaSR1_AaSR2_AaSR3_AaSR4_AaSR5_AaSR6_AaSR7_AaSR8_AaSR9_AaSR10_AaSR11_r12_lFyAA0F6ParserCKcfU_ + 4518
	16  sourcery                            0x00000001013e7657 $S9Commander7commandyAA11CommandType_px_q_q0_q1_q2_q3_q4_q5_q6_q7_q8_q9_q10_q11_y05ValueD0Qz_ADQy_ADQy0_ADQy1_ADQy2_ADQy3_ADQy4_ADQy5_ADQy6_ADQy7_ADQy8_ADQy9_ADQy10_ADQy11_tKctAA18ArgumentDescriptorRzAaSR_AaSR0_AaSR1_AaSR2_AaSR3_AaSR4_AaSR5_AaSR6_AaSR7_AaSR8_AaSR9_AaSR10_AaSR11_r12_lFyAA0F6ParserCKcfU_TA + 279
	17  sourcery                            0x00000001013d4343 $S9Commander16AnonymousCommandVAA0C4TypeA2aDP3runyyAA14ArgumentParserCKFTW + 51
	18  sourcery                            0x00000001013d4793 $S9Commander11CommandTypePAAE3runys5NeverOSSSgF + 403
	19  sourcery                            0x000000010147b78a $S8Sourcery6runCLIyyF + 2314
	20  sourcery                            0x000000010147a1fd main + 61
	21  libdyld.dylib                       0x00007fff6ea913d5 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
Abort trap: 6
```

To me it looks like the file is just not generated.

**Comment by amichnia** [comment]
@torstenlehmann  could you give a bit more details?

**Comment by torstenlehmann** [comment]
Already described here: https://github.com/krzysztofzablocki/Sourcery/issues/745

Don't have further info.

**Comment by fmirault** [comment]
@amichnia Thank you for your answer. I tried your 1 and 2, and I got the exact same error as @torstenlehmann. 

I'm using Xcode 10.2 (10E125), Sourcery (0.16.0) from Podfile.lock, cocoapods 1.6.1, and the command line I tried was `Pods/Sourcery/bin/Sourcery.app/Contents/MacOS/Sourcery --config mocky.yml --disableCache` and also `Pods/Sourcery/bin/sourcery --config mocky.yml --disableCache`

**Comment by amichnia** [comment]
@torstenlehmann @fmirault @maxmamis

If you are using older SwiftyMocky with Sourcery 0.16.0, there is an issue with `arguments` being renamed to `argument`. I was getting similar errors.

SwiftyMocky 3.2.0 is now released, it works on top of Sourcery 0.16.0, with both Swift 4.2 and 5.0. Should fix it 🤞
